### PR TITLE
Translate state and lifecycle page

### PR DESCRIPTION
**Pull request by jhonmike**
#1 

Sections:
- [x] Intro to State and Lifecycle
- [x] Converting a Function to a Class
- [x] Adding Local State to a Class
- [x] Adding Lifecycle Methods to a Class
- [x] Using State Correctly
- [x] The Data Flows Down

### CODE REVIEW

**Comment by tibuurcio** [comment]
Hi @jhonmike, thank you for taking the time to contribute to translating the docs 👍 

Just to give you a heads up, I think as per #4 we shouldn't translate _state_ to _estado_ since it refers to such an important concept in React.

**Comment by cezaraugusto** [comment]
@tibuurcio what do you think about using `estado (_state_)`? I personally like this way

**Comment by tibuurcio** [comment]
@cezaraugusto I personally think leaving just `state` is better, but I don't think `estado (state)` is bad in any way. Also, you have a lot more [experience](https://github.com/cezaraugusto/You-Dont-Know-JS/graphs/contributors) in this than I do 😆. 

Maybe I'm just used to see it being written as `state` in english articles that it feels strange to translate such a big concept in the React's API.

**Comment by jhonmike** [comment]
Also I was thinking, I think it could be "state" only.

**Comment by halian-vilela** [comment]
> @tibuurcio what do you think about using `estado (_state_)`? I personally like this way

I think that state can represent two different things and should be treated acordingly...

When representing the current state of the application, as a snapshot in time, I don't think it loses semantic value if we translate it to "estado".

I guess it only loses value when it's talking about `state` as a property of react and all that its concerned with. In this case I think we could use it without translation.

Makes sense?

**Comment by jhonmike** [comment]
So what's the decision?

**Comment by tibuurcio** [comment]
@halian-vilela It makes sense, indeed :)

File: content/docs/state-and-lifecycle.md [hunk 1]
```diff
@@ -409,41 +409,41 @@ Then you can update them independently with separate `setState()` calls:
...
 function FormattedDate(props) {
   return <h2>It is {props.date.toLocaleTimeString()}.</h2>;
 }
 ```
 
-[**Try it on CodePen**](http://codepen.io/gaearon/pen/zKRqNB?editors=0010)
+[**Experimente no CodePen**](http://codepen.io/gaearon/pen/zKRqNB?editors=0010)
 
-This is commonly called a "top-down" or "unidirectional" data flow. Any state is always owned by some specific component, and any data or UI derived from that state can only affect components "below" them in the tree.
+Isso é comumente chamado de fluxo de dados "top-down" ou "unidirecionala". Qualquer state é sempre de propriedade de algum componente específico, e qualquer dado ou interface do usuário derivado desse state só pode afetar os componentes "abaixo" deles na árvore.
```
Comment (LeoMosiah) [inline content/docs/state-and-lifecycle.md]:
```suggestion
Isso é comumente chamado de fluxo de dados "top-down" ou "unidirecional". Qualquer state é sempre de propriedade de algum componente específico, e qualquer dado ou interface do usuário derivado desse state só pode afetar os componentes "abaixo" deles na árvore.
```

File: content/docs/state-and-lifecycle.md [hunk 2]
```diff
@@ -302,84 +302,84 @@ ReactDOM.render(
...
 ```js
-// Correct
+// Correto
 this.setState((state, props) => ({
   counter: state.counter + props.increment
 }));
 ```
 
-We used an [arrow function](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Functions/Arrow_functions) above, but it also works with regular functions:
+Usamos um [arrow function](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Functions/Arrow_functions) acima, mas também funciona com funções regulares:
```
Comment (halian-vilela) [inline content/docs/state-and-lifecycle.md]:
```suggestion
Usamos uma [arrow function](https://developer.mozilla.org/pt-BR/docs/Web/JavaScript/Reference/Functions/Arrow_functions) acima, mas também funciona com funções regulares:
```


Função é feminino. Creio que é melhor utilizar "uma" _arrow function_

Comment (glaucia86) [reply content/docs/state-and-lifecycle.md]: Concordo @halian-vilela. Pode alterar. Sem problemas.

**Comment by jhonmike** [comment]
Esse `force` foi apenas um update da master com um `--rebase` no final :)

File: content/docs/state-and-lifecycle.md [hunk 3]
```diff
@@ -193,19 +193,19 @@ ReactDOM.render(
...
 ```
 
-[**Try it on CodePen**](http://codepen.io/gaearon/pen/KgQpJd?editors=0010)
+[**Experimente no CodePen**](http://codepen.io/gaearon/pen/KgQpJd?editors=0010)
 
-Next, we'll make the `Clock` set up its own timer and update itself every second.
+Em seguinda, faremos a configuração do próprio temporizador e atualizaremos a cada segundo.
 
-## Adding Lifecycle Methods to a Class {#adding-lifecycle-methods-to-a-class}
+## Adicionando métodos de ciclo de vida a classe {#adding-lifecycle-methods-to-a-class}
```
Comment (cezaraugusto) [inline content/docs/state-and-lifecycle.md]: pode fazer uma revisão nos títulos e deixá-los todos em capital como no original? exemplo: `Adicionando Métodos de Ciclo de Vida a Classe`

**Comment by jhonmike** [comment]
Ajustado @cezaraugusto

**Pull request merged**